fix: account for BEACON_ROOTS_ADDRESS access

DavePearce opened the pull request:
It seems that, post CANCUN, both SSTORE and SLOAD invoke the system transaction and that this results in the BEACON_ROOTS_ADDRESS being accessed.  The existing tests did not account for this.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Add fork-aware helper for system-transaction address count and adjust HubShomei tests to include it.
> 
> - **Core**:
>   - Add `Fork.numberOfAddressesSeenBySystemTransaction(fork)` returning fork-specific counts (`0` pre-Cancun, `1` Cancun, `2` Prague).
> - **Tests**:
>   - Update `HubShomeiTests` to import `Fork` and adjust `addressSeen.size()` assertions to include `Fork.numberOfAddressesSeenBySystemTransaction(fork)`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 32f11a392eef9f98c5c4151285d1dd704461f836. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

DavePearce committed:
fix: account for BEACON_ROOTS_ADDRESS access

It seems that, post CANCUN, both SSTORE and SLOAD invoke the system
transaction and that this results in the BEACON_ROOTS_ADDRESS being
accessed.  The existing tests did not account for this.

letypequividelespoubelles reviewed (commented):
IMHO, the cleaner would be to have a static method 'nbOfAddressSeenBySystenTransaction()' in Fork.java that returns 0 for London to Shanghai, 1 for Cancun, 2 for Prague and Osaka, and in the assert have
NbAddressSeen = 4 + nbOfAddressSeenBySystenTransaction(testInfo.firk)

DavePearce committed:
support Prague for addresses seen by system tx

This adds support for the number of addresses seen in Prague by the
system transaction.
